plugins/woocommerce/client/blocks/assets/js/blocks/product-specifications/edit.tsx

```diff
@@ -117,60 +121,113 @@ const Edit = ( {
 		);
 	}
 
-	const productData: Record< string, { label: string; value: string } > = {
-		weight: {
+	const productData: Record< string, { label: string; value: string } > = {};
+
+	if ( showWeight ) {
+		productData.weight = {
 			label: __( 'Weight', 'woocommerce' ),
 			value: '',
-		},
-		dimensions: {
+		};
+
+		if ( isSpecificProductContext ) {
+			productData.weight.value = product.weight
+				? `${ product.weight } ${ weightUnit }`
+				: '';
+		} else {
+			productData.weight.value = `10 ${ weightUnit }`;
+		}
+
+		productData.weight.value =
+			isSpecificProductContext && product.weight
+				? `${ product.weight } ${ weightUnit }`
+				: `10 ${ weightUnit }`;
```

Comment:
Seems like leftover and duplication of lines 132-138 (without the case of empty string when there's no product data)